torchvision/models/googlenet.py

```diff
@@ -0,0 +1,183 @@
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from torch.utils import model_zoo
+
+__all__ = ['GoogLeNet', 'googlenet']
+
+model_urls = {
+    # GoogLeNet ported from TensorFlow
+    'googlenet': 'https://github.com/TheCodez/vision/releases/download/1.0/googlenet-1378be20.pth',
+}
+
+
+def googlenet(pretrained=False, **kwargs):
+    r"""GoogLeNet (Inception v1) model architecture from
+    `"Going Deeper with Convolutions" <http://arxiv.org/abs/1409.4842>`_.
+    Args:
+        pretrained (bool): If True, returns a model pre-trained on ImageNet
+    """
+    if pretrained:
+        if 'transform_input' not in kwargs:
+            kwargs['transform_input'] = True
+        kwargs['init_weights'] = False
+        model = GoogLeNet(**kwargs)
+        model.load_state_dict(model_zoo.load_url(model_urls['googlenet']))
+        return model
+
+    return GoogLeNet(**kwargs)
+
+
+class GoogLeNet(nn.Module):
+
+    def __init__(self, num_classes=1000, aux_logits=True, transform_input=False, init_weights=True):
+        super(GoogLeNet, self).__init__()
+        self.aux_logits = aux_logits
+        self.transform_input = transform_input
+
+        self.conv1 = BasicConv2d(3, 64, kernel_size=7, stride=2, padding=3)
+        self.maxpool1 = nn.MaxPool2d(3, stride=2, ceil_mode=True)
+        self.conv2 = BasicConv2d(64, 64, kernel_size=1)
+        self.conv3 = BasicConv2d(64, 192, kernel_size=3, padding=1)
+        self.maxpool2 = nn.MaxPool2d(3, stride=2, ceil_mode=True)
+
+        self.inception3a = Inception(192, 64, 96, 128, 16, 32, 32)
+        self.inception3b = Inception(256, 128, 128, 192, 32, 96, 64)
+        self.maxpool3 = nn.MaxPool2d(3, stride=2, ceil_mode=True)
+
+        self.inception4a = Inception(480, 192, 96, 208, 16, 48, 64)
+        self.inception4b = Inception(512, 160, 112, 224, 24, 64, 64)
+        self.inception4c = Inception(512, 128, 128, 256, 24, 64, 64)
+        self.inception4d = Inception(512, 112, 144, 288, 32, 64, 64)
+        self.inception4e = Inception(528, 256, 160, 320, 32, 128, 128)
+        self.maxpool4 = nn.MaxPool2d(3, stride=2, ceil_mode=True)
```

Comment:
@fmassa one thing to note here is that TensorFlow uses 2x2 pooling instead of 3x3. Don't know if that has a positive impact on the accuracy, but it would mean to further diverge from the paper.

Reply:
What was the reason why you used 3x3 pooling, in order to make everything work out fine, given the differences between TF and PyTorch?

Reply:
This didn't cause problems during my conversion process so I probably just missed it. Should I change it? In that case it might be a good idea to add a note that the implementation differs from the paper.

Reply:
let me try seeing if it makes a difference for the performance, and I'll let you know

Reply:
Accuracy is 1 point better using the 2x2 pooling, with
```
Acc@1 69.778 Acc@5 89.530
```
so I'll be changing it. Thanks for the heads up